data/transactions/transaction.go

```diff
@@ -649,3 +650,41 @@ func (tc ExplicitTxnContext) GenesisID() string {
 func (tc ExplicitTxnContext) GenesisHash() crypto.Digest {
 	return tc.GenHash
 }
+
+// SignedTxGroup used as the in-memory representation of a signed transaction group.
+// unlike the plain array of signed transactions, this includes transaction origination and counter
+// used by the transaction pool and the transaction sync
+//msgp: ignore SignedTxGroup
+type SignedTxGroup struct {
```

Comment:
It's true that it's unique to the transaction sync and the transaction pool.. it's a shared data structure, and as such, it would probably be a good fit for `/data/...`. It might be argued that it would be better off in a separate package under `/data`, which I'm not opposing to.
Keep in mind that the node architecture is so that the transaction pool and the transaction sync aren't familiar with each other. The node is the one that bridges them - and so these data structures need to be visible to all the intermediate components as well.